### PR TITLE
feature/FLW-1688 systematic solution

### PR DESCRIPTION
**Pull request by Deneath**

### CODE REVIEW

File: app/src/main/java/jp/co/soramitsu/app/root/presentation/RootViewModel.kt [hunk 1]
```diff
@@ -19,19 +26,44 @@ class RootViewModel(
...
+        checkAppVersion()
         interactor.runBalancesUpdate()
             .onEach { handleUpdatesSideEffect(it) }
             .launchIn(this)
 
         updatePhishingAddresses()
     }
 
+    private fun checkAppVersion() = viewModelScope.launch {
+        val appVersion = getAppVersion().replace(".", "")
```
Comment (PankraSerg) [inline app/src/main/java/jp/co/soramitsu/app/root/presentation/RootViewModel.kt:58]: why replace "."? 2.1.11 and 2.11.1 will have no difference

Comment (Deneath) [reply app/src/main/java/jp/co/soramitsu/app/root/presentation/RootViewModel.kt:58]: right, will fix

File: app/src/main/java/jp/co/soramitsu/app/root/presentation/main/MainViewModel.kt [hunk 2]
```diff
@@ -1,19 +1,40 @@
...
+    companion object {
+        private const val IDLE_MINUTES: Long = 20
+    }
 
     init {
         externalRequirements.value = ChainConnection.ExternalRequirement.ALLOWED
     }
 
+    fun onScreenAppears() {
+        startDropSessionTimer()
```
Comment (PankraSerg) [inline app/src/main/java/jp/co/soramitsu/app/root/presentation/main/MainViewModel.kt:28]: is the timer resets if user hides and opened app again? or moved to pin in time from first timer start

File: common/src/main/java/jp/co/soramitsu/common/domain/GetAppVersion.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+class GetAppVersion(private val contextManager: ContextManager) {
+    operator fun invoke() = contextManager.getContext().let {
+        val versionName = it.packageManager.getPackageInfo(it.packageName, 0).versionName
+        val regex = Regex("[^0-9]")
+        regex.replace(versionName, "")
+    }
+}
+
+fun isAppVersionSupported(minSupportedVersion: String?, appVersionName: String): Boolean {
+    val appVersion = appVersionName.replace(".", "").toInt()
```
Comment (PankraSerg) [inline common/src/main/java/jp/co/soramitsu/common/domain/GetAppVersion.kt:14]: not a good idea 2.1.11 and 2.11.1 also 2.0.11 = 2011 is greater then 2.1.3 = 213 which is not in terms of app versions

File: feature-account-impl/src/main/java/jp/co/soramitsu/feature_account_impl/presentation/account/details/AccountDetailsViewModel.kt [hunk 4]
```diff
@@ -103,15 +116,19 @@ class AccountDetailsViewModel(
         val accountIcon = projection?.let {
             iconGenerator.createAddressIcon(it.accountId, AddressIconGenerator.SIZE_SMALL, backgroundColorRes = R.color.account_icon_dark)
         } ?: resourceManager.getDrawable(R.drawable.ic_warning_filled)
-
+        Log.d("&&&", "========================================================================")
+        Log.d("&&&", accountInChain.chain.minSupportedVersion.toString())
+        Log.d("&&&", appVersion)
+        Log.d("&&&", isAppVersionSupported(accountInChain.chain.minSupportedVersion, appVersion).toString())
```
Comment (PankraSerg) [inline feature-account-impl/src/main/java/jp/co/soramitsu/feature_account_impl/presentation/account/details/AccountDetailsViewModel.kt:122]: better remove

File: feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/di/WalletFeatureModule.kt [hunk 5]
```diff
@@ -234,4 +243,8 @@ class WalletFeatureModule {
     @Provides
     @FeatureScope
     fun provideSelectedFiatUseCase(preferences: Preferences) = SelectedFiat(preferences)
+
+//    @Provides
+//    @FeatureScope
+//    fun provideAppVersionUseCase(contextManager: ContextManager) = GetAppVersion(contextManager)
```
Comment (PankraSerg) [inline feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/di/WalletFeatureModule.kt:249]: remove

File: feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/presentation/balance/list/BalanceListAdapter.kt [hunk 6]
```diff
@@ -99,6 +99,10 @@ class AssetViewHolder(
     fun bind(model: AssetWithStateModel, itemHandler: BalanceListAdapter.ItemAssetHandler) {
         val asset = model.asset
 
+        if (asset.isSupported.not()) { // update to get more from the app
```
Comment (PankraSerg) [inline feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/presentation/balance/list/BalanceListAdapter.kt:102]: remove comment

**Review by PankraSerg — APPROVED** [review]

**Review by PankraSerg — APPROVED** [review]

**Pull request merged**